### ISSUE TITLE
Wrong diagnostic output when using `cfg_attr` and derive-macro-helper-attributes

### ISSUE THREAD

**Issue by Forsworns** [rust-analyzer/rust-analyzer#10736]
I'm using stable `rustc 1.56.0 (09c42c458 2021-10-18)` and `rust-analyzer v0.2.809`. I found that derive-macro-helper-attributes have to be placed before related derive-macros when using `cfg_attr`. Otherwise, the rust-analyzer will generate red highlights indicating there exist errors. 

The problem is related to `cfg_attr`, I made a minimal example so that you can verify it locally.

```rust
// main.rs
#[cfg(feature = "use_kube_rs")]
use kube::CustomResource;
use schemars::JsonSchema;
use serde::{Deserialize, Serialize};

#[cfg_attr(
    feature = "use_kube_rs",
    kube(group = "clux.dev", version = "v1", kind = "Foo", namespaced),
    derive(CustomResource, Clone, Debug, Default, PartialEq, Deserialize, Serialize, JsonSchema)
)]
struct FooSpec {
    info: String,
}


#[cfg_attr(
    feature = "use_kube_rs",
    derive(CustomResource, Clone, Debug, Default, PartialEq, Deserialize, Serialize, JsonSchema),
    kube(group = "clux.dev", version = "v1", kind = "Bar", namespaced)
)]
struct BarSpec {
    info: String,
}

fn main() {
    println!("Hello, world!");
}
```

```toml
# Cargo.toml

[package]
name = "test_cfg_attr"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[features]
default = ["use_kube_rs"]
use_kube_rs = ["kube", "k8s-openapi"]

[dependencies]
kube = {version="0.63.2", features = ["default","derive", "runtime"], optional=true}
k8s-openapi = {version="0.13.1", features = ["v1_22"], optional=true}
schemars = "0.8.6"
serde = { version = "1.0.126", features = ["derive"]}
serde_json = "1.0.64"
```

Here are screenshot for illustrations:

![image](https://user-images.githubusercontent.com/24567080/141092235-4dd7a551-c78e-463b-b905-bd39472d30b4.png)

![image](https://user-images.githubusercontent.com/24567080/141092133-6f38342e-b68d-422a-8531-e594b6cf0991.png)

![image](https://user-images.githubusercontent.com/24567080/141092205-55bba181-9ab3-4cd2-9afc-71916fec6913.png)

I'm not sure but it seems a bug...

**Comment by Veykril** [comment]
Duplicate of https://github.com/rust-analyzer/rust-analyzer/issues/10110, for now you can split up the `cfg_attr` in two, that is make one per inner attribute as a workaround to get rid of the errors.

**Comment by nyurik** [comment]
I tried this approach per @Veykril for #15840, but that didn't work :(

```rust
#[cfg_attr(feature = "display", derive(displaydoc::Display))]
#[cfg_attr(feature = "display", ignore_extra_doc_attributes)]
```
